ServantPersistent/src/DBApi.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE DataKinds       #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TypeOperators   #-}
+{-# LANGUAGE TypeApplications   #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilyDependencies #-}
+{-# LANGUAGE TypeInType #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DeriveAnyClass #-}
+module DBApi
+    where
+
+import Data.Aeson
+import Data.Aeson.TH
+import Database.Persist.Sql
+import Database.Persist.TH
+import GHC.Generics
+import Data.Time.Clock
+import Data.Time
+import Data.Text
+import Types
+import Control.Monad.IO.Class
+import Control.Monad.Reader
+import Control.Lens
+import Data.ByteString
+import Data.Proxy
+import Models
+import Safe
+
+type TenantOutput = TenantBase TenantOutputField
+
+data TenantOutputField = TOF { status :: TenantStatus
+                             , ownerId :: Maybe DBUserId
+                             } deriving (Generic, ToJSON)
+
+tbdbIso :: Iso' (TenantBase TenantOutputField) DBTenant
```

Comment:
Is this isomorphism only for tenants, or are you expecting to use this as a pattern for every model? Will it work for `User` who's output will never contain the password field.